nncf/torch/tensor_statistics/collectors.py

```diff
@@ -9,31 +9,41 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import Any, Callable, Deque, List, Optional, Union
+from functools import partial
+from typing import Deque, List, Optional, Tuple, Union
 
+import numpy as np
 import torch
 
-from nncf.common.tensor import NNCFTensor
 from nncf.common.tensor import TensorElementsType
-from nncf.common.tensor_statistics.collectors import MeanMinMaxStatisticCollector
-from nncf.common.tensor_statistics.collectors import MeanPercentileStatisticCollector
-from nncf.common.tensor_statistics.collectors import MeanStatisticCollector
-from nncf.common.tensor_statistics.collectors import MedianMADStatisticCollector
-from nncf.common.tensor_statistics.collectors import MinMaxStatisticCollector
-from nncf.common.tensor_statistics.collectors import MixedMinMaxStatisticCollector
+from nncf.common.tensor_statistics.collectors import MaskedReduceFN
 from nncf.common.tensor_statistics.collectors import NNCFCollectorTensorProcessor
-from nncf.common.tensor_statistics.collectors import PercentileStatisticCollector
-from nncf.common.tensor_statistics.collectors import ReductionShape
-from nncf.common.tensor_statistics.reduction import np_percentile_reduce_like
-from nncf.torch.dynamic_graph.context import no_nncf_trace
+from nncf.common.tensor_statistics.collectors import NNCFTensor
+from nncf.experimental.common.tensor_statistics.collectors import AbsMaxReducer
+from nncf.experimental.common.tensor_statistics.collectors import AbsQuantileReducer
+from nncf.experimental.common.tensor_statistics.collectors import BatchMeanReducer
+from nncf.experimental.common.tensor_statistics.collectors import MaxAggregator
+from nncf.experimental.common.tensor_statistics.collectors import MaxReducer
+from nncf.experimental.common.tensor_statistics.collectors import MeanAggregator
+from nncf.experimental.common.tensor_statistics.collectors import MeanPerChReducer
+from nncf.experimental.common.tensor_statistics.collectors import MeanReducer
+from nncf.experimental.common.tensor_statistics.collectors import MedianAbsoluteDeviationAggregator
+from nncf.experimental.common.tensor_statistics.collectors import MinAggregator
+from nncf.experimental.common.tensor_statistics.collectors import MinReducer
+from nncf.experimental.common.tensor_statistics.collectors import NoopReducer
+from nncf.experimental.common.tensor_statistics.collectors import PrecentileAggregator
+from nncf.experimental.common.tensor_statistics.collectors import QuantileReducer
+from nncf.experimental.common.tensor_statistics.collectors import ShapeAggregator
+from nncf.experimental.common.tensor_statistics.collectors import TensorCollector
```

Comment:
Can we use only one import instead of 15?

Reply:
Also, I've just noticed that we keep `reducers` and `collectors` in one file. Is it correct placement?

Reply:
This may not be possible to do while preserving autolinter functionality (`black`/`ruff`). Anyway, when browsing code one does not really look at this list at all, instead we just scroll down to where the actual code begins.

I agree that the aggregators and reducers should have the code in respective separate files.

Reply:
We can separate them but I don't see a problem to keep them in one file

Reply:
Can we do it in a separate PR, please? I believe it is enough refactoring for this PR. I will create ticket on it